add clap for argument parsing

kbknapp opened the pull request.

brinchj commented:
(FYI: Don't worry about Travis, it hasn't been able to match a nightly with the code successfully yet)

brinchj commented:
I see the call to license now :-)

Let's disable that for now at least, it's too verbose. What I was referring to in the email was that "--license" should eventually include the licenses of dependencies too, but it currently doesn't (I was remembering that wrong, which is why the email made little sense wrt code).

brinchj commented:
Wrt the NAME and PATH arguments, I think it just gets printed wrong due to something in arg_from_usage (\<ATH\> instead of \<PATH\>).

brinchj commented:
Be aware that another pull requests, and a few of my own commits, has gone in too. None of them touch main.rs though.

kbknapp commented:
I've made those changes, and everything appears to working. Let me know if see any issues! :)

Quick summary of changes:
* Updated Cargo.lock (clap is v0.6.2, everything is parsing correctly)
* Removed `license()` call from empty run i.e. `hat-backup` with no arguments
* Added clap license (MIT) as LICENSE-CLAP
* Added printing of CLAP license in `license()`

brinchj commented:
Can you rebase this commit on top of master?
I don't think there's been any changes to main, but there's been a change to CONTRIBUTORS that may need merging.

kbknapp commented:
Will do!

kbknapp commented:
@brinchj I rebased onto master, everything seems to be running good to me. Let me know if you see anything.

brinchj commented:
What compiler did you use to build the changes?
Did you upgrade to newest nightly?

brinchj commented on src/hat/main.rs:
```diff
@@ -77,82 +84,88 @@ static MAX_BLOB_SIZE: usize = 4 * 1024 * 1024;
-fn usage() {
-  println!("Usage: {} [snapshot|commit|checkout] name path", env::args().next().unwrap());
-}
+// #[cfg(not(test))]
```
let's remove any commented out code.

brinchj commented on src/hat/main.rs:
```diff
@@ -42,13 +42,20 @@ extern crate sqlite3;
 
 
-use std::env;
+// use std::env;
```
let's remove any commented out code.

kbknapp commented:
Commented code has been removed.

I used the latest nightly to compile and test. `rustc 1.0.0-nightly (00978a987 2015-04-18) (built 2015-04-19)`

brinchj commented:
Trying this out now, and will merge if working :+1:
